src/components/Reports/FourteenMonthReports/useFourteenMonthReport.ts

```diff
@@ -0,0 +1,127 @@
+import { useMemo } from 'react';
+import {
+  GetFourteenMonthReportQuery,
+  useGetFourteenMonthReportQuery,
+} from 'src/components/Reports/FourteenMonthReports/GetFourteenMonthReport.generated';
+import { convertStatus } from 'src/utils/functions/convertContactStatus';
+import { FourteenMonthReportCurrencyType } from './FourteenMonthReport';
+
+type CurrencyGroups = NonNullable<
+  GetFourteenMonthReportQuery['reportsSalaryCurrencyDonations']
+>['currencyGroups'];
+
+export const useFourteenMonthReport = (
+  accountListId: string,
+  currencyType: FourteenMonthReportCurrencyType,
+  designationAccounts?: string[],
+) => {
+  const {
+    data: gqlData,
+    loading,
+    error,
+  } = useGetFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+      // Backend sends one extra month, e.g. 13m returns 14 months
+      range: '13m',
+      designationAccountId:
+        designationAccounts && designationAccounts.length > 0
+          ? designationAccounts
+          : null,
+    },
+  });
+
+  const data =
+    currencyType === FourteenMonthReportCurrencyType.Salary
+      ? gqlData?.reportsSalaryCurrencyDonations
+      : gqlData?.reportsDonorCurrencyDonations;
+
+  const fourteenMonthReport = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+
+    const isSalaryType =
+      currencyType === FourteenMonthReportCurrencyType.Salary;
+    const currencyGroups: CurrencyGroups = data.currencyGroups;
+
+    return {
+      currencyType,
+      salaryCurrency: data.defaultCurrency ?? 'USD',
+      currencyGroups: Object.entries(currencyGroups ?? {}).map(
+        ([currency, currencyGroup]: [string, CurrencyGroups[string]]) => ({
+          currency,
+          totals: {
+            year: Number(
+              isSalaryType
+                ? currencyGroup.totals.year_converted
+                : currencyGroup.totals.year,
+            ),
+            months: currencyGroup.totals.months.map((total, index) => ({
+              month: data.months[index],
+              total: Number(total),
+            })),
+            average: currencyGroup.donation_infos.reduce(
+              (averageTotal, contactDonationInfo) =>
+                averageTotal + Number(contactDonationInfo.average),
+              0,
+            ),
+            minimum: currencyGroup.donation_infos.reduce(
+              (minimumTotal, contactDonationInfo) =>
+                minimumTotal + Number(contactDonationInfo.minimum),
+              0,
+            ),
+          },
+          contacts: currencyGroup.donation_infos
+            .map((contactDonationInfo) => {
+              const contact = data.donorInfos.find(
```

Comment:
I don't think we would want to show rows where we couldn't find a contact, right? Since most of the fields will be empty. Can we return `null` if we don't find a contact and then filter out `null` values before sorting?